src/services/suggestionDiagnostics.ts

```diff
@@ -17,12 +17,27 @@ namespace ts {
             switch (node.kind) {
                 case SyntaxKind.FunctionDeclaration:
                 case SyntaxKind.FunctionExpression:
+
                     if (isJsFile) {
                         const symbol = node.symbol;
                         if (symbol.members && (symbol.members.size > 0)) {
                             diags.push(createDiagnosticForNode(isVariableDeclaration(node.parent) ? node.parent.name : node, Diagnostics.This_constructor_function_may_be_converted_to_a_class_declaration));
                         }
                     }
+                    
+                    if(node.modifiers && containsAsync(node.modifiers)){
+                        break;
+                    }
+
+                    const returnType = checker.getReturnTypeOfSignature(checker.getSignatureFromDeclaration(<FunctionDeclaration | FunctionExpression> node))

```

Comment:
limit these to non-ambient functions. an easy check is if the function does not have a body, break

Reply:
what about methods, and arrow functions?